### PR TITLE
Remove walrus operator to read code in linear flow

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Fixed a section that was obviously difficult to read due to assignment expressions

@adeak Thanks for your advice.

### CODE REVIEW

**Comment by user27182** [comment]
Probably a good style rule to avoid use of the walrus operator with ternary if-else blocks. They are indeed a bit confusing to read since the variable must be assigned in the middle of the expression. 

I tried looking for a rule with ruff or similar to enforce this but couldn't find anything.

**Review by user27182 — DISMISSED** [review]

File: doc/source/make_tables.py [hunk 1]
```diff
@@ -406,7 +406,8 @@ def get_row(cls, i, row_data):
 
 def _get_doc(func: Callable[[], Any]) -> Optional[str]:
     """Return the first line of the callable's docstring."""
-    return doc.splitlines()[0] if (doc := func.__doc__) else None
+    doc = func.__doc__
+    return doc.splitlines()[0] if (doc) else None
```
Comment (adeak) [inline doc/source/make_tables.py:410]: Parentheses were only necessary for the assignment expression:
```suggestion
    return doc.splitlines()[0] if doc else None
```

File: pyvista/examples/_dataset_loader.py [hunk 2]
```diff
@@ -81,8 +81,9 @@ def num_files(self) -> int:
 
         If a path is a folder, the number of files contained in the folder is returned.
         """
-        path = [path] if isinstance(path := self.path, str) else path
-        return sum([1 if os.path.isfile(p) else len(_get_all_nested_filepaths(p)) for p in path])
+        path = self.path
+        paths = [path] if isinstance(path, str) else path
+        return sum([1 if os.path.isfile(p) else len(_get_all_nested_filepaths(p)) for p in paths])
```
Comment (adeak) [inline pyvista/examples/_dataset_loader.py:86]: Unrelated, but we can use a generator comprehension here
```suggestion
        return sum(1 if os.path.isfile(p) else len(_get_all_nested_filepaths(p)) for p in paths)
```

**Review by adeak — CHANGES_REQUESTED** [review]

**Comment by adeak** [comment]
> Probably a good style rule to avoid use of the walrus operator with ternary if-else blocks. They are indeed a bit confusing to read since the variable must be assigned in the middle of the expression.
> 
> I tried looking for a rule with ruff or similar to enforce this but couldn't find anything.

My recommendation would be to restrict the (mis)feature to the header line of conditional blocks or loops, as a general rule. If there's some consensus about this we could add it to our contributing guide.

**Review by adeak — APPROVED** [review]

**Review by user27182 — APPROVED** [review]

**Pull request merged**